### PR TITLE
Fixed RTL text not beeing rendered properly on Lib images

### PR DESCRIPTION
**Pull request by JPVenson**
Building upon the work of #8050 and discussion in chat, using several character groups to evaluate if the text drawn on library images should be handled LTR or RTL

**Changes**
- Added RTL detection for following text cultures: 
  - Arabic
  - Armenian
  - Hebrew
  - Syriac
  - Thaana

**Issues**
#9610

### CODE REVIEW

File: src/Jellyfin.Drawing.Skia/StripCollageBuilder.cs [hunk 1]
```diff
@@ -22,6 +23,9 @@ public StripCollageBuilder(SkiaEncoder skiaEncoder)
         _skiaEncoder = skiaEncoder;
     }
 
+    [GeneratedRegex("(/s{IsArabic}||/s{IsArmenian}||/s{IsHebrew}||/s{IsSyriac}||/s{IsThaana})+", RegexOptions.IgnoreCase)]
```
Comment (crobibero) [inline src/Jellyfin.Drawing.Skia/StripCollageBuilder.cs:26]:
```suggestion
    [GeneratedRegex("\p{IsArabic}|\p{IsArmenian}|\p{IsHebrew}|\p{IsSyriac}|\p{IsThaana}", RegexOptions.IgnoreCase)]
```

I think the regex supplied is incorrect

Also, I don't think `IgnoreCase` is required for these character groups? Though to be fair I don't know anything about RTL languages

Comment (Khaled0Ebrahim) [reply src/Jellyfin.Drawing.Skia/StripCollageBuilder.cs:26]: > I think the regex supplied is incorrect
> 
> Also, I don't think `IgnoreCase` is required for these character groups? Though to be fair I don't know anything about RTL languages

Yes, You are right there is no case for rtl languages

Comment (JPVenson) [reply src/Jellyfin.Drawing.Skia/StripCollageBuilder.cs:26]: I made some additional tests and there is something strange going one with testing for multiple groups at once. Will need more time to test that.

Comment (Khaled0Ebrahim) [reply src/Jellyfin.Drawing.Skia/StripCollageBuilder.cs:26]: Also, make sure to remove the hard-coded width

Comment (crobibero) [reply src/Jellyfin.Drawing.Skia/StripCollageBuilder.cs:26]: I think if you move to the regex I suggested you would have more success

Comment (Khaled0Ebrahim) [reply src/Jellyfin.Drawing.Skia/StripCollageBuilder.cs:26]: > I think if you move to the regex I suggested you would have more success

I agree

Comment (JPVenson) [reply src/Jellyfin.Drawing.Skia/StripCollageBuilder.cs:26]:
```
Microsoft (R) Visual C# Interactive Compiler version 4.4.0-6.22559.4 ()
Loading context from 'CSharpInteractive.rsp'.
Type "#help" for more information.
> var regex = new System.Text.RegularExpressions.Regex(@"\p{IsArabic}||\p{IsArmenian}||\p{IsHebrew}||\p{IsSyriac}||\p{IsThaana}");
> regex.IsMatch("test")
true
> regex.IsMatch("بْجَدِيَّة الْعَرَ")
true
> 
```
Nah not really. As i said, weird stuff happening i have to test beforehand.

Comment (JPVenson) [reply src/Jellyfin.Drawing.Skia/StripCollageBuilder.cs:26]: Dont mind me, i cannot read.

Comment (Khaled0Ebrahim) [reply src/Jellyfin.Drawing.Skia/StripCollageBuilder.cs:26]: I geuss it is related to the way you provided that arabic I can give you arabic words to test

Comment (Khaled0Ebrahim) [reply src/Jellyfin.Drawing.Skia/StripCollageBuilder.cs:26]: wait wait or is written like this in regex | not double

Comment (Khaled0Ebrahim) [reply src/Jellyfin.Drawing.Skia/StripCollageBuilder.cs:26]: copy and paste the text crobibero wrote

Comment (Khaled0Ebrahim) [reply src/Jellyfin.Drawing.Skia/StripCollageBuilder.cs:26]: you can try this
bool hasRTL = Regex.IsMatch("نص باللغة العربيه", @"\p{IsArabic}|\p{IsHebrew}|\p{IsSyriac}|\p{IsThaana}");

Comment (Khaled0Ebrahim) [reply src/Jellyfin.Drawing.Skia/StripCollageBuilder.cs:26]: you xan try this
`bool hasRTL = Regex.IsMatch("نص باللغة العربيه", @"\p{IsArabic}|\p{IsHebrew}|\p{IsSyriac}|\p{IsThaana}");`

Comment (crobibero) [reply src/Jellyfin.Drawing.Skia/StripCollageBuilder.cs:26]: It works perfectly fine if using the Regex I sent.

https://dotnetfiddle.net/YkYPxB

Comment (JPVenson) [reply src/Jellyfin.Drawing.Skia/StripCollageBuilder.cs]: > Dont mind me, i cannot read.

File: src/Jellyfin.Drawing.Skia/StripCollageBuilder.cs [hunk 2]
```diff
@@ -22,6 +23,9 @@ public StripCollageBuilder(SkiaEncoder skiaEncoder)
         _skiaEncoder = skiaEncoder;
     }
 
+    [GeneratedRegex(@"\s{IsArabic}|\s{IsArmenian}|\s{IsHebrew}|\s{IsSyriac}|\s{IsThaana}", RegexOptions.IgnoreCase)]
```
Comment (crobibero) [inline src/Jellyfin.Drawing.Skia/StripCollageBuilder.cs:26]:
```suggestion
    [GeneratedRegex(@"\p{IsArabic}|\p{IsArmenian}|\p{IsHebrew}|\p{IsSyriac}|\p{IsThaana}")]
```

Change `\s` to `\p`
Remove `RegexOptions.IgnoreCase` as it has no effect.

Comment (Khaled0Ebrahim) [reply src/Jellyfin.Drawing.Skia/StripCollageBuilder.cs]: And remove the hard-coded width of 250

**Comment by Khaled0Ebrahim** [comment]
Great, just remove the hard-coded 250
and make the text centered

**Review by Shadowghost — APPROVED** [review]

**Comment by crobibero** [comment]
image

This looks correct to me, so I'm going to merge it. If we still have issues follow-up PRs can be created.

**Review by crobibero — APPROVED** [review]

**Pull request merged**